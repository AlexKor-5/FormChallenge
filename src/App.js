import React from 'react'
import { Step1 } from './pages/Step1/Step1'
import { Formik, Form } from 'formik'
import * as Yup from 'yup'
import { MainContainer } from './components/MainContainer/MainContainer'
import { Title } from './components/Title/Title'
import AccountCircleRoundedIcon from '@mui/icons-material/AccountCircleRounded'
import { TextInput } from './components/TextInput/TextInput'
import { MyButton } from './components/MyButton/MyButton'
import { PhoneNumberInput } from './components/PhoneNumberInput/PhoneNumberInput'

// const validate = values => {
//     const errors = {}
//     if (!values.email) {
//         errors.email = 'Required'
//     } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
//         errors.email = 'Invalid email address'
//     }
//     return errors
// }

export const App = () => {
    return (
        <>
            <Formik
                initialValues={{
                    firstName: '',
                    lastName: '',
                    email: '',
                    phoneNumber: '',
                }}
                validationSchema={Yup.object({
                    firstName: Yup.string()
                        .max(15, 'Have to be 15 characters or less')
                        .required('Required'),
                    lastName: Yup.string()
                        .max(20, 'Have to be 20 or less characters')
                        .required('Required'),
                    email: Yup.string().email('Invalid email.').required('Required.'),
                })}
                // validate={validate}
                onSubmit={(values, { setSubmitting }) => {
                    setTimeout(() => {
                        alert(JSON.stringify(values, null, 2))
                        setSubmitting(false)
                    }, 400)
                }}
            >
                <Form>
                    <MainContainer>
                        <Title text={'Step 2'} iconRender={<AccountCircleRoundedIcon />} />
                        <TextInput text={'Email'} name={'email'} />
                        {/*<TextInput text={'Phone Number'} name={'phoneNumber'} />*/}
                        <PhoneNumberInput text={'Phone Number'} name={'phoneNumber'} />
                        <MyButton>Next</MyButton>
                    </MainContainer>
                </Form>
            </Formik>
        </>
    )
}
