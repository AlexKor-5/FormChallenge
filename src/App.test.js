import React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import { fireEvent } from '@testing-library/react'
import { App } from './App'
import { BrowserRouter } from 'react-router-dom'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'

//https://testing-playground.com/ -> helper to find a right role of element in HTML

describe('App', () => {
    beforeEach(() => {
        render(
            <BrowserRouter>
                <App />
            </BrowserRouter>
        )
    })

    it('calls submit', async () => {
        await waitFor(() =>
            fireEvent.change(getFirstName(), { target: { value: 'Alexander' } })
        )
        expect(getFirstName()).toHaveValue('Alexander')

        await waitFor(() =>
            fireEvent.change(getLastName(), { target: { value: 'Kor' } })
        )
        expect(getLastName()).toHaveValue('Kor')

        // const email = await screen.findByRole('textbox', {
        //     name: /email/i,
        // })
        // console.log(email)

        // fireEvent.change(await findEmail(), {
        //     target: { value: '218855@ukr.net' },
        // })
        // expect(await findEmail()).toHaveValue('218855@ukr.net')
    })
})

const getFirstName = () =>
    screen.getByRole('textbox', {
        name: /first name/i,
    })

const getLastName = () =>
    screen.getByRole('textbox', {
        name: /last name/i,
    })
const findEmail = () => screen.findByLabelText(/email/i)
