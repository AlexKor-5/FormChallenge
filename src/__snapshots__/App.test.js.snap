// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders First Name correctly 1`] = `
<input
  aria-invalid="false"
  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
  id="mui-1"
  name="firstName"
  type="text"
  value=""
/>
`;

exports[`App renders First Name correctly 2`] = `
<input
  aria-invalid="false"
  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
  id="mui-1"
  name="firstName"
  type="text"
  value="Alexander"
/>
`;
